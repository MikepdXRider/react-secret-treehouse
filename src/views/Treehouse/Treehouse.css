.signOut {
  background-color: #ffce01;
  border: 1px solid #fcba12;
  color: #f24501;
  font-size: 1.2rem;
  padding: 7px 15px;
  text-transform: uppercase;
  font-weight: bold;
  margin: 1.5rem 0;
}
