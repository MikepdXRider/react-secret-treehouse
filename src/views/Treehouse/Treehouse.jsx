import { Link, useHistory } from 'react-router-dom';
import { useAuth } from '../../hooks/useAuth';
import secret from '../../assets/top-secret.png';
import styles from './Treehouse.css';

export default function Treehouse() {
  const history = useHistory();
  const { user, logout } = useAuth();

  const handleLogout = () => {
    logout(() => history.push('/'));
  };

  return (
    <>
      <h1>Welcome to the Secret Treehouse!</h1>
      <p>You are signed in as {user.email}</p>
      <Link to="/treehouse/members">View the Treehouse Members</Link>
      <button
        aria-label="Sign Out"
        type="button"
        className={styles.signOut}
        onClick={handleLogout}
      >
        Sign out
      </button>
      <img src={secret} alt="Top secret treehouse image" />
    </>
  );
}
