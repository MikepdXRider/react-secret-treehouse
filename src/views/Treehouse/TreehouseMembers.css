.members {
  max-width: 60vw;
}
