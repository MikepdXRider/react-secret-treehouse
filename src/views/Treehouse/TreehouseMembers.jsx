import { Link } from 'react-router-dom';
import styles from './TreehouseMembers.css';
import members from '../../assets/secret-treehouse-members.jpeg';

export default function TreehouseMembers() {
  return (
    <>
      <Link to="/treehouse">Go back to the Treehouse</Link>
      <img src={members} alt="Top secret treehouse members" className={styles.members} />
      <ul>
        <li>John</li>
        <li>Paul</li>
        <li>George</li>
        <li>Ringo</li>
      </ul>
    </>
  );
}
