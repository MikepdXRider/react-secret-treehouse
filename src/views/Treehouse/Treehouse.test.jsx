import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import Treehouse from './Treehouse';

const mockLogout = jest.fn();

jest.mock('../../hooks/useAuth', () => {
  return {
    useAuth: function () {
      return {
        user: { email: 'test@example.com' },
        logout: mockLogout,
      };
    },
  };
});

// Snapshot test
it('should render the treehouse homepage if logged in', () => {
  const { container } = render(
    <MemoryRouter>
      <Treehouse />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

// Behavior test
it('should call the logout function when clicking the sign out button', () => {
  render(
    <MemoryRouter>
      <Treehouse />
    </MemoryRouter>
  );

  const signOutButton = screen.getByRole('button', { name: 'Sign Out' });

  userEvent.click(signOutButton);
  expect(mockLogout).toHaveBeenCalledTimes(1);
});
