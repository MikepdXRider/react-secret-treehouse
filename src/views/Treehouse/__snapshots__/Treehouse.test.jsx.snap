// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the treehouse homepage if logged in 1`] = `
<div>
  <h1>
    Welcome to the Secret Treehouse!
  </h1>
  <p>
    You are signed in as 
    test@example.com
  </p>
  <a
    href="/treehouse/members"
  >
    View the Treehouse Members
  </a>
  <button
    aria-label="Sign Out"
    class="signOut"
    type="button"
  >
    Sign out
  </button>
  <img
    alt="Top secret treehouse image"
    src="/path/image"
  />
</div>
`;
